components/RecipeDetails.js

```diff
@@ -2,25 +2,33 @@ import styled from "styled-components";
 import Image from "next/image";
 import Link from "next/link";
 
-export default function RecipeDetails({ recipeData }) {
+export default function RecipeDetails({ recipeDetails }) {
   return (
     <RecipeCard>
-      <h1> {recipeData.name}</h1>
+      <h1> {recipeDetails.name}</h1>
       <StyledImage
-        src={recipeData.imageURL}
-        alt={recipeData.name}
+        src={
+          recipeDetails.imageURL
+            ? recipeDetails.imageURL
+            : "/recipe-images/generic-food-3.png"
+        }
+        alt={recipeDetails.name}
         width={300}
         height={300}
       />
-      <p>⏰ {recipeData.preparationTime}</p>
+      <p>⏰ {recipeDetails.preparationTime}</p>
       <ul>
-        {recipeData.ingredients.map((ingredient) => (
-          <li key={ingredient.ingredient}>
-            {ingredient.ingredient} - {ingredient.quantity}
-          </li>
-        ))}
+        {recipeDetails.ingredients.map((ingredient) =>
+          ingredient.ingredient ? (
+            <li key={ingredient.ingredient}>
+              {ingredient.quantity} - {ingredient.ingredient}
+            </li>
+          ) : (
+            ""
+          )
+        )}
```

Comment:
```suggestion
        {recipeDetails.ingredients.map((ingredient) =>
          ingredient.ingredient &&
            <li key={ingredient.ingredient}>
              {ingredient.quantity} - {ingredient.ingredient}
            </li>
          
        )}
```

this is also kind of confusing to map over ingredients and then have ingredient.ingredient.. I would expect it to be ingredient.name at that point.

Reply:
Thank you thats brilliant!